base/sparse/linalg.jl

```diff
@@ -800,6 +826,15 @@ scale{T,Tv,Ti}(b::Vector{T}, A::SparseMatrixCSC{Tv,Ti}) =
 function factorize(A::SparseMatrixCSC)
     m, n = size(A)
     if m == n
+        if istril(A)
+            if istriu(A)
+                return return Diagonal(A)
```

Comment:
No. I also thought it would be an error, but then realized that the line actually has test coverage. However, a few other lines in `factorize(SparseMatrixCSC)` don't so I've prepared a small commit with extra tests which also removes the extra return.